### PR TITLE
update MessageType signature

### LINKED ISSUE #28160: Shoudl MessageType of message be changed?
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
code: 
```
1, tsx
2, code like `message.warn('bla...').then(() => {// some code here})`
```
It report an error: 
```
TS2554: Expected 2 arguments, but got 1. An argument for 'reject' was not provided.
```
caused by the signature of MessageType:
```
export interface MessageType {
    (): void;
    then: (fill: ThenableArgument, reject: ThenableArgument) => Promise<void>;
    promise: Promise<void>;
}
```

### What does the proposed API look like?
```
// I copy Promise.then signature in Webstorm lib.es5.d.ts
export interface MessageType {
    (): void;
    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): PromiseLike<TResult1 | TResult2>;
    promise: Promise<void>;
}
```
or I think the reject is not needed, cause message doesn't have reject action.

### PR DESCRIPTION
**Pull request by maicss**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve https://github.com/ant-design/ant-design/issues/28160

### 💡 Background and solution
described in linked issue
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  change MessageType interface |
| 🇨🇳 Chinese |  修改 MessageTyupe 的接口 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/message/index.tsx [hunk 1]
```diff
@@ -108,7 +108,10 @@ export interface ThenableArgument {
 
 export interface MessageType {
   (): void;
-  then: (fill: ThenableArgument, reject: ThenableArgument) => Promise<void>;
+  then<TResult1 = T, TResult2 = never>(
+    onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null,
+    onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null
+  ): Promise<TResult1 | TResult2>;
```
Comment (afc163) [inline components/message/index.tsx:114]: Could you add test case for TS compile like https://github.com/ant-design/ant-design/blob/master/components/form/__tests__/type.test.tsx ?

Comment (maicss) [reply components/message/index.tsx:114]: I donot how to test a interface...do you mean code like:

https://github.com/ant-design/ant-design/blob/273ab186daaa945e5ac2e00913f969004d6eefdb/components/form/__tests__/type.test.tsx#L10-L21

Comment (maicss) [reply components/message/index.tsx:114]: Just assert some jsx `tobeTruthy`? And should I delete reject signature of then?

Comment (afc163) [reply components/message/index.tsx:114]: Add test case written in TypeScript, make sure it won't break when compile it.

Comment (maicss) [reply components/message/index.tsx:114]: @afc163 算了，我不提这个pr了。
第一，这个很简单，你们自己copy一下代码就好了，或者全any。
第二，我问了两个问题，第二个是：是不是应该把then的第二个参数改成可选或者直接删掉，因为message是没有交互的，最终一定是成功的。可能是我英语太差的缘故，导致你没看懂，你也没回答。

Comment (afc163) [reply components/message/index.tsx:114]: > 是不是应该把then的第二个参数改成可选或者直接删掉

这个可以有。

Comment (afc163) [reply components/message/index.tsx:114]: 看看 https://github.com/ant-design/ant-design/pull/28304 是不是能解决问题？

Comment (maicss) [reply components/message/index.tsx:114]: 可以的

**Pull request closed without merging**